### ISSUE TITLE
Javascript AJAX or fetch Calls are failing

### ISSUE THREAD

**Issue by drveresh** [brave/brave-browser#33096]
## Description 
Basic AJAX or Javascript fetch() calls are failing, see the attachment.
<img width="773" alt="01" src="https://github.com/brave/brave-browser/assets/3812468/8c82babf-58d7-401b-991c-3e4fa17a8ad4">

## Steps to Reproduce
Run the below code in the browser console in the Inspector:
```
async function fetchGitHubUserInfo() {
    const response = await fetch("https://my-json-server.typicode.com/typicode/demo/db");
    const data = await response.json()
    console.log(data);
};fetchGitHubUserInfo();
```

## Actual result:
Uncaught (in promise) SyntaxError: Unexpected end of JSON input

## Expected result:
Valid-JOSN-Payload

## Reproduces how often: 
100% times issue is reproduciblele

## Brave version (brave://version info)
Mac Laptop (iOS 13.5.1 (22G90))
[Version 1.14.81 Chromium: 85.0.4183.102 (Official Build) (64-bit)](https://brave.com/latest/)

## Other Additional Information:
Same issue with lot of other endpoints like: `https://httpstat.us/200`

**Comment by kenherbert** [comment]
Also fails in Chrome, but not Firefox.

Can you please link an endpoint that isn't affected? This might be due to a CORS thing, and having a working endpoint may help prove/disprove it.

**Comment by drveresh** [comment]
@kenherbert No, it is working fine with my Chrome (`Version 116.0.5845.140 (Official Build) (x86_64)`), even with all other endpoints, refer the screenshot.  I tested with several of my production service endpoints, both with and without CORS, they all worked fine with all browsers, except Brave.
![image](https://github.com/brave/brave-browser/assets/3812468/b127018b-997f-465a-8064-b877865c3e1d)

![image](https://github.com/brave/brave-browser/assets/3812468/2291be4c-f284-42a9-8a47-1d86629efead)

![image](https://github.com/brave/brave-browser/assets/3812468/a3e921f4-f3bb-4243-b071-9b0fd2fe7808)

![image](https://github.com/brave/brave-browser/assets/3812468/e90acf6e-6e4e-48bf-9a54-22c43c047640)

**Comment by kenherbert** [comment]
Further testing (using Brave 1.58.127, Chrome 117.0.5938.89, Edge 117.0.2045.31)

Brave from new tab page: Failed - Failed to fetch error
Chrome from new tab page: Failed - Failed to fetch error
Edge from new tab page: Rejected - Content Security Policy error

Brave from github.com - Rejected - Content Security Policy error
Chrome from github.com - Rejected - Content Security Policy error
Edge from github.com - Rejected - Content Security Policy error

Brave from google.com - Success
Chrome from google.com - Success
Edge from google.com - Success

Brave from https://my-json-server.typicode.com/typicode/demo/db - Success
Chrome from https://my-json-server.typicode.com/typicode/demo/db - Success
Edge from https://my-json-server.typicode.com/typicode/demo/db - Rejected - Content Security Policy error

Brave and Chrome are showing consistent successes and failures for me. Edge on the other hand.....
